adapters/pubmatic.go

```diff
@@ -44,38 +49,103 @@ type pubmaticParams struct {
 		return nil, err
 	}
 
+	adSlotFlag := false
+	pubId := ""
+	if len(bidder.AdUnits) > MAX_IMPRESSIONS_PUBMATIC {
+		glog.Warningf("[PUBMATIC] First %d impressions will be considered from request tid %s\n",
+			MAX_IMPRESSIONS_PUBMATIC, pbReq.ID)
+	}
+
 	for i, unit := range bidder.AdUnits {
 		var params pubmaticParams
 		err := json.Unmarshal(unit.Params, &params)
 		if err != nil {
-			return nil, err
+			glog.Warningf(PrepareLogMessage(pbReq.ID, params.PublisherId, unit.Code, unit.BidID,
+				fmt.Sprintf("Ignored bid: invalid JSON  [%s] err [%s]", unit.Params, err.Error())))
+			continue
 		}
+
 		if params.PublisherId == "" {
-			return nil, errors.New("Missing publisherId param")
+			glog.Warningf(PrepareLogMessage(pbReq.ID, params.PublisherId, unit.Code, unit.BidID,
+				fmt.Sprintf("Ignored bid: Publisher Id missing")))
+			continue
 		}
+		pubId = params.PublisherId
 		if params.AdSlot == "" {
-			return nil, errors.New("Missing adSlot param")
+			glog.Warningf(PrepareLogMessage(pbReq.ID, params.PublisherId, unit.Code, unit.BidID,
+				fmt.Sprintf("Ignored bid: adSlot missing")))
+			continue
+		}
+
+		adSlotStr := strings.TrimSpace(params.AdSlot)
+		adSlot := strings.Split(adSlotStr, "@")
+		if len(adSlot) == 2 && adSlot[0] != "" && adSlot[1] != "" {
+
+			if pbReq.Imp[i].Banner != nil {
+				pbReq.Imp[i].Banner.Format = nil // pubmatic doesn't support
+				adSize := strings.Split(strings.ToLower(strings.TrimSpace(adSlot[1])), "x")
+				if len(adSize) == 2 {
+					width, err := strconv.Atoi(strings.TrimSpace(adSize[0]))
+					if err != nil {
+						glog.Warningf(PrepareLogMessage(pbReq.ID, params.PublisherId, unit.Code, unit.BidID,
+							fmt.Sprintf("Ignored bid: invalid adSlot width [%s]", adSize[0])))
+						continue
+					}
+
+					heightStr := strings.Split(strings.TrimSpace(adSize[1]), ":")
+					height, err := strconv.Atoi(strings.TrimSpace(heightStr[0]))
+					if err != nil {
+						glog.Warningf(PrepareLogMessage(pbReq.ID, params.PublisherId, unit.Code, unit.BidID,
+							fmt.Sprintf("Ignored bid: invalid adSlot height [%s]", heightStr[0])))
+						continue
+					}
+
+					pbReq.Imp[i].TagID = strings.TrimSpace(adSlot[0])
+					pbReq.Imp[i].Banner.H = uint64(height)
+					pbReq.Imp[i].Banner.W = uint64(width)
+					adSlotFlag = true
+				} else {
+					glog.Warningf(PrepareLogMessage(pbReq.ID, params.PublisherId, unit.Code, unit.BidID,
+						fmt.Sprintf("Ignored bid: invalid adSize [%s]", adSize)))
+					continue
+				}
+			}
+		} else {
+			glog.Warningf(PrepareLogMessage(pbReq.ID, params.PublisherId, unit.Code, unit.BidID,
+				fmt.Sprintf("Ignored bid: invalid adSlot [%s]", params.AdSlot)))
+			continue
 		}
-		if pbReq.Imp[i].Banner != nil {
-			pbReq.Imp[i].Banner.Format = nil
-		} // pubmatic doesn't support
-		pbReq.Imp[i].TagID = params.AdSlot
+
 		if pbReq.Site != nil {
-			pbReq.Site.Publisher = &openrtb.Publisher{ID: params.PublisherId}
+			siteCopy := *pbReq.Site
```

Comment:
@dbemiller Changes are merged from #140

Reply:
git/github is still showing merge conflicts from that line though... the conflicts need to be resolved before we can merge it.

Reply:
Actually... nevermind. I thought you had to do this because I don't have write permissions to your repo, but github lets me resolve conflicts directly in the browser.

Looks good to me now.